app-creator/client/src/utils/template-generation.ts

```diff
@@ -194,6 +194,33 @@ export function transferData() {
         }
       }
 
+      /**
+       * Getting remaining widgets that have not been added. We will populate the scratch panel
+       * with these widgets so that the user can add them later on.
+       */
+      const remainingWidgetIds = [];
+      for (const id in widgets) {
+        if (
+          id !== SCRATCH_PANEL &&
+          !id.startsWith('panel') &&
+          !id.startsWith('sidemenu') &&
+          !(id in store.getState().template.widgets)
```

Comment:
It's better to wrap this up into a helper method, `isActiveWidget(widget)`, since these conditions may need to be updated.